SignalServiceKit/src/Messages/OWSReadReceiptManager.m

```diff
@@ -568,11 +569,22 @@ - (BOOL)areReadReceiptsEnabled
 
 - (void)setAreReadReceiptsEnabled:(BOOL)value
 {
-    DDLogInfo(@"%@ areReadReceiptsEnabled: %d.", self.tag, value);
```

Comment:
Looks like `areReadReceiptsEnabled` can open a transaction inside a transaction. See how `OWSMessageManager` uses it, for example.  I think we may need to load this cached value explicitly on app launch.

Reply:
I've also wary of introducing deadlocks in this manager as we did with the portfolio manager.